api/cloudprotocol/unitstatus.go

```diff
@@ -0,0 +1,148 @@
+	StatesPartition   = "states"
+	ServicesPartition = "services"
+	LayersPartition   = "layers"
+)
+
+/***********************************************************************************************************************
+ * Types
+ **********************************************************************************************************************/
+
+// UnitConfigStatus unit config status.
+type UnitConfigStatus struct {
+	Version   string     `json:"version"`
+	Status    string     `json:"status"`
+	ErrorInfo *ErrorInfo `json:"errorInfo,omitempty"`
+}
+
+// CPUInfo cpu information.
+type CPUInfo struct {
+	ModelName  string `json:"modelName"`
+	NumCores   uint64 `json:"totalNumCores"`
+	NumThreads uint64 `json:"totalNumThreads"`
+	Arch       string `json:"arch"`
+	ArchFamily string `json:"archFamily"`
+	MaxDMIPs   uint64 `json:"maxDmips"`
+}
+
+// PartitionInfo partition information.
+type PartitionInfo struct {
+	Name      string   `json:"name"`
+	Types     []string `json:"types"`
+	TotalSize uint64   `json:"totalSize"`
+}
+
+// NodeInfo node information.
+type NodeInfo struct {
+	NodeID     string          `json:"nodeId"`
+	NodeType   string          `json:"nodeType"`
+	Name       string          `json:"name"`
+	Status     string          `json:"status"`
+	CPUs       []CPUInfo       `json:"cpus"`
+	OSTypes    string          `json:"osTypes"`
+	MaxDMIPs   uint64          `json:"maxDmips"`
+	TotalRAM   uint64          `json:"totalRam"`
+	Attrs      []string        `json:"attrs,omitempty"`
+	Partitions []PartitionInfo `json:"partitions,omitempty"`
+}
+
+// ServiceStatus service status.
+type ServiceStatus struct {
+	ServiceID string     `json:"serviceId"`
+	Version   string     `json:"version"`
+	Status    string     `json:"status"`
+	ErrorInfo *ErrorInfo `json:"errorInfo,omitempty"`
+}
+
+// InstanceStatus service instance runtime status.
+type InstanceStatus struct {
+	aostypes.InstanceIdent
+	Version       string     `json:"version"`
+	StateChecksum string     `json:"stateChecksum,omitempty"`
+	RunState      string     `json:"runState"`
+	NodeID        string     `json:"nodeId"`
+	ErrorInfo     *ErrorInfo `json:"errorInfo,omitempty"`
+}
+
+// LayerStatus layer status.
+type LayerStatus struct {
+	LayerID   string     `json:"layerId"`
+	Digest    string     `json:"digest"`
+	Version   string     `json:"version"`
+	Status    string     `json:"status"`
+	ErrorInfo *ErrorInfo `json:"errorInfo,omitempty"`
+}
+
+// ComponentStatus component status.
+type ComponentStatus struct {
+	ComponentID   string     `json:"componentId"`
+	ComponentType string     `json:"componentType"`
+	Version       string     `json:"version"`
+	Status        string     `json:"status"`
+	ErrorInfo     *ErrorInfo `json:"errorInfo,omitempty"`
+}
+
+// UnitStatus unit status structure.
+type UnitStatus struct {
+	MessageType  string             `json:"messageType"`
+	IsDeltaInfo  bool               `json:"isDeltaInfo"`
+	UnitConfig   []UnitConfigStatus `json:"unitConfig"`
```

Comment:
UnitConfig seems to be a single value

Reply:
No. We may return multiple items for unit config status:

`[{version: 1, "installed"},{ version: 2, "error"}]`

To indicate that current version 1, installing version 2 fails.